### PR TITLE
Frontend first  submission

### PR DESCRIPTION
**Pull request by c007456**
This is the first time that the front-end web project submits pr

### CODE REVIEW

File: frontend/babel.config.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// 这是项目发布阶段需要用到的 babel 插件
```
Comment (lanking520) [inline frontend/babel.config.js:1]:
```suggestion
// This is the necessary babel plugin for project publishing
```

File: frontend/src/App.vue [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<template>
+  <div id="app">
+    <!-- 路由占位符 -->
```
Comment (lanking520) [inline frontend/src/App.vue:3]:
```suggestion
    <!-- Placeholder for routing -->
```

File: frontend/src/assets/css/common.less [hunk 3]
```diff
@@ -0,0 +1,11 @@
+/* 全局变量 */
```
Comment (lanking520) [inline frontend/src/assets/css/common.less:1]:
```suggestion
/* Global variable */
```

File: frontend/src/assets/css/common.less [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+// 正文字体大小
```
Comment (lanking520) [inline frontend/src/assets/css/common.less:2]:
```suggestion
// body font size
```

File: frontend/src/assets/css/common.less [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+@textSize:14px; 
+// 标题字体大小
```
Comment (lanking520) [inline frontend/src/assets/css/common.less:4]:
```suggestion
// Title font size
```

File: frontend/src/fliter.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+// 导入格式化时间的插件
```
Comment (lanking520) [inline frontend/src/fliter.js:1]:
```suggestion
// time formatter plugin
```

File: frontend/src/fliter.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+import moment from 'moment'
+// 定义全局的过滤器
```
Comment (lanking520) [inline frontend/src/fliter.js:3]:
```suggestion
// Global filter on time
```

File: frontend/src/main.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
+import Vue from 'vue'
+import App from './App.vue'
+import router from './router'
+import '@/plugins/element.js'
+//过滤器
```
Comment (lanking520) [inline frontend/src/main.js:5]:
```suggestion
```

File: frontend/src/main.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+import * as filters from "./fliter"
+// vuex
+import store from './store'
+
+// 导入字体图标
```
Comment (lanking520) [inline frontend/src/main.js:10]:
```suggestion
// Import font
```

File: frontend/src/main.js [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+// import './assets/fonts/iconfont.css'
+// 导入全局样式表
```
Comment (lanking520) [inline frontend/src/main.js:12]:
```suggestion
// import global css
```

File: frontend/src/store.js [hunk 11]
```diff
@@ -0,0 +1,20 @@
+// 注册 vuex
```
Comment (lanking520) [inline frontend/src/store.js:1]:
```suggestion
// register vuex
```

File: frontend/src/store.js [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+import Vuex from 'vuex'
+import Vue from 'vue'
+
+Vue.use(Vuex)
+
+var store = new Vuex.Store({
+  state: { // this.$store.state.***
+   
+  },
+  mutations: { // this.$store.commit('方法的名称', '按需传递唯一的参数')
```
Comment (lanking520) [inline frontend/src/store.js:11]:
```suggestion
  mutations: { // this.$store.commit('method name', 'provide if necessary')
```

File: frontend/src/view/Log.vue [hunk 13]
```diff
@@ -0,0 +1,106 @@
+<template>
+  <div class="log">
+    <h3>Log</h3>
+    <div class="log-box">
+      <div class="title">Running Log</div>
+      <div class="log-content">
+        <pre>
+        [INFO ] - Starting djl-serving: 0.18.0 ...
```
Comment (lanking520) [inline frontend/src/view/Log.vue:8]: Why do we keep these?

**Review by lanking520 — COMMENTED** [review]
Thanks for your contribution, a few comments

**Review by frankfliu — COMMENTED** [review]
Please push to a different branch in your repo, so we can push our changes to that branch.

File: frontend/README.md [hunk 14]
```diff
@@ -0,0 +1,20 @@
+DJLServing background management system
```
Comment (frankfliu) [inline frontend/README.md:1]:
```suggestion
# DJLServing management console
```

File: frontend/README.md [hunk 15]
```diff
@@ -0,0 +1,20 @@
...
+# djl-serving-web
```
Comment (frankfliu) [inline frontend/README.md:2]:
```suggestion
```

File: frontend/README.md [hunk 16]
```diff
@@ -0,0 +1,20 @@
...
+
+## Project setup
```
Comment (frankfliu) [inline frontend/README.md:4]: Leave a empty line before code block
```suggestion
## Project setup

```

File: frontend/README.md [hunk 17]
```diff
@@ -0,0 +1,20 @@
+DJLServing background management system
+# djl-serving-web
+
+## Project setup 
+```
+npm install
+```
+
+### Compiles and hot-reloads for development
```
Comment (frankfliu) [inline frontend/README.md:9]:
```suggestion
### Compiles and hot-reloads for development

```

File: frontend/README.md [hunk 18]
```diff
@@ -0,0 +1,20 @@
...
+```
+npm run serve
+```
+
+### Compiles and minifies for production
```
Comment (frankfliu) [inline frontend/README.md:14]:
```suggestion
### Compiles and minifies for production

```

File: serving/src/main/java/ai/djl/serving/ServerInitializer.java [hunk 19]
```diff
@@ -12,11 +12,7 @@
  */
 package ai.djl.serving;
 
-import ai.djl.serving.http.ConfigurableHttpRequestHandler;
-import ai.djl.serving.http.InferenceRequestHandler;
-import ai.djl.serving.http.InvalidRequestHandler;
-import ai.djl.serving.http.KServeRequestHandler;
-import ai.djl.serving.http.ManagementRequestHandler;
+import ai.djl.serving.http.*;
```
Comment (frankfliu) [inline serving/src/main/java/ai/djl/serving/ServerInitializer.java:15]: Avoid use *

File: serving/src/main/java/ai/djl/serving/http/LogRequestHandler.java [hunk 20]
```diff
@@ -0,0 +1,209 @@
...
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/** A class handling inbound HTTP requests for the log API. */
+public class LogRequestHandler extends HttpRequestHandler {
+
+    private static final Logger logger = LoggerFactory.getLogger(LogRequestHandler.class);
```
Comment (frankfliu) [inline serving/src/main/java/ai/djl/serving/http/LogRequestHandler.java:40]: Please run formatJava and pass all static checks:

```
cd serving
./gradlew check -x test
```

File: serving/src/main/java/ai/djl/serving/http/LogRequestHandler.java [hunk 21]
```diff
@@ -0,0 +1,209 @@
...
+    private static final String LOG_PATH = "/logs";
+
+    private RequestParser requestParser;
```
Comment (frankfliu) [inline serving/src/main/java/ai/djl/serving/http/LogRequestHandler.java:43]:
```suggestion
```

File: serving/src/main/java/ai/djl/serving/http/LogRequestHandler.java [hunk 22]
```diff
@@ -0,0 +1,209 @@
...
+
+    private static final Pattern PATTERN = Pattern.compile("^/logs([/?].*)?");
+
+    /** default constructor. */
```
Comment (frankfliu) [inline serving/src/main/java/ai/djl/serving/http/LogRequestHandler.java:47]: this constructor can be removed

File: serving/src/main/java/ai/djl/serving/http/LogRequestHandler.java [hunk 23]
```diff
@@ -0,0 +1,209 @@
...
+
+    /** {@inheritDoc} */
+    @Override
+    protected void handleRequest(
+            ChannelHandlerContext ctx,
+            FullHttpRequest req,
+            QueryStringDecoder decoder,
+            String[] segments)
+            throws ModelException {
+        HttpMethod method = req.method();
```
Comment (frankfliu) [inline serving/src/main/java/ai/djl/serving/http/LogRequestHandler.java:70]:
```
        if (!HttpMethod.GET.equals(req.method())) {
            throw new MethodNotAllowedException();
        }
        String modelServerHome = ConfigManager.getModelServerHome();
        Path dir = Paths.get(modelServerHome, "logs");
        if (segments.length < 3) {
            listLogs(ctx, dir);
        } else if (segments.length <= 4) {
            String fileName = segments[2];
            if (segments.length == 4 && "download".equals(segments[3])) {
                downloadLog(ctx, dir, fileName);
            } else {
                int lines = NettyUtils.getIntParameter(decoder, "lines", 200);
                showLog(ctx, dir, fileName, lines);
            }
        } else {
            throw new ResourceNotFoundException();
        }
```

File: serving/src/main/java/ai/djl/serving/http/LogRequestHandler.java [hunk 24]
```diff
@@ -0,0 +1,209 @@
...
+        } else if (segments.length == 3) {
+            String fileName = segments[2];
+            handleLogText(ctx, fileName, decoder, dir);
+        } else if (segments.length == 4 && "download".equals(segments[2])) {
+            String fileName = segments[3];
+            handleLogDownload(ctx, dir, fileName);
+        }
+    }
+
+    private void handleLogDownload(ChannelHandlerContext ctx, Path dir, String fileName)
```
Comment (frankfliu) [inline serving/src/main/java/ai/djl/serving/http/LogRequestHandler.java:87]:
```
    private void downloadLog(ChannelHandlerContext ctx, Path dir, String fileName) {
        if (fileName.contains("..")) {
            throw new BadRequestException("Invalid log file name:" + fileName);
        }
        Path file = dir.resolve(fileName);
        if (!Files.isRegularFile(file)) {
            throw new BadRequestException("File does not exist");
        }
        try (InputStream is = Files.newInputStream(file)) {
            HttpResponseStatus status = HttpResponseStatus.OK;
            FullHttpResponse resp =
                    new DefaultFullHttpResponse(HttpVersion.HTTP_1_1, status, false);
            String name = URLEncoder.encode(fileName, "UTF-8");
            String contentDisposition = "attachment;fileName=" + name + ";fileName*=UTF-8''" + name;
            resp.headers()
                    .set("Content-Type", "text/plain")
                    .set("Content-Disposition", contentDisposition);

            byte[] buf = new byte[8192];
            int read;
            while ((read = is.read(buf)) != -1) {
                resp.content().writeBytes(buf, 0, read);
            }
            NettyUtils.sendHttpResponse(ctx, resp, true);
        } catch (IOException e) {
            throw new InternalServerException("Failed to read log file: " + fileName);
        }
    }
```

File: serving/src/main/java/ai/djl/serving/http/LogRequestHandler.java [hunk 25]
```diff
@@ -0,0 +1,209 @@
...
+        } finally {
+            try {
+                input.close();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    private void handleLogList(ChannelHandlerContext ctx, Path dir) {
```
Comment (frankfliu) [inline serving/src/main/java/ai/djl/serving/http/LogRequestHandler.java:133]:
```
    private void listLogs(ChannelHandlerContext ctx, Path dir) {
        if (!Files.isDirectory(dir)) {
            NettyUtils.sendJsonResponse(ctx, Collections.emptyList());
            return;
        }

        List<Map<String, String>> list = new ArrayList<>();
        try (Stream<Path> stream = Files.walk(dir)) {
            stream.forEach(
                    f -> {
                        File file = f.toFile();
                        String fileName = file.getName();
                        if (fileName.endsWith(".log")) {
                            Map<String, String> m = new ConcurrentHashMap<>(3);
                            m.put("name", fileName);
                            m.put("lastModified", String.valueOf(file.lastModified()));
                            m.put("length", String.valueOf(file.length()));
                            list.add(m);
                        }
                    });
        } catch (IOException e) {
            throw new InternalServerException("Failed to list log files");
        }
        NettyUtils.sendJsonResponse(ctx, list);
    }
```

File: serving/src/main/java/ai/djl/serving/http/LogRequestHandler.java [hunk 26]
```diff
@@ -0,0 +1,209 @@
...
+                                    m.put("lastModified", v.lastModified());
+                                    m.put("length", v.length());
+                                    return m;
+                                })
+                        .collect(Collectors.toList());
+
+        NettyUtils.sendJsonResponse(ctx, list);
+    }
+
+    private void handleLogText(
```
Comment (frankfliu) [inline serving/src/main/java/ai/djl/serving/http/LogRequestHandler.java:156]:
```
    private void showLog(ChannelHandlerContext ctx, Path dir, String fileName, int lines) {
        if (fileName.contains("..")) {
            throw new BadRequestException("Invalid log file name:" + fileName);
        }
        Path file = dir.resolve(fileName);
        if (!Files.isRegularFile(file)) {
            throw new BadRequestException("File does not exist");
        }

        String lastLineText = getLastLineText(file.toFile(), lines);
        NettyUtils.sendJsonResponse(ctx, lastLineText);
    }

    private String getLastLineText(File file, int lines) {
        long fileLength = file.length() - 1;
        if (file.length() < 0) {
            return "";
        }

        StringBuilder builder = new StringBuilder();
        try (RandomAccessFile raf = new RandomAccessFile(file, "r")) {
            int readLines = 0;
            raf.seek(fileLength);
            for (long pointer = fileLength; pointer >= 0; pointer--) {
                raf.seek(pointer);
                char c;
                c = (char) raf.read();
                if (c == '\n') {
                    readLines++;
                    if (readLines == lines) break;
                }
                builder.append(c);
                fileLength = fileLength - pointer;
            }
            builder.reverse();
        } catch (IOException e) {
            throw new InternalServerException("Failed to read log file.");
        }
        return builder.toString();
    }
```

Comment (frankfliu) [reply serving/src/main/java/ai/djl/serving/http/LogRequestHandler.java]: I moved frontend into plugins/manage-console folder.

Everything is working before you merge the branch. Please take a look and let's discuss offline.

**Review by frankfliu — APPROVED** [review]

**Pull request merged**